x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/operator/lookup/QueryList.java

```diff
@@ -70,7 +71,12 @@ public static QueryList rawTermQueryList(MappedFieldType field, SearchExecutionC
             }
             case BYTES_REF -> offset -> {
                 BytesRefBlock bytesRefBlock = (BytesRefBlock) block;
-                return bytesRefBlock.getBytesRef(offset, new BytesRef());
+                BytesRef value = bytesRefBlock.getBytesRef(offset, new BytesRef());
+                if (field.typeName().equals("text")) {
+                    // Text fields involve case-insensitive contains queries, we need to use lowercase on the term query
+                    return new BytesRef(value.utf8ToString().toLowerCase(Locale.ROOT));
```

Comment:
This mostly exists to show that matching against text fields "doesn't work like you think it does" in the ENRICH infrastructure we've borrowed.

Reply:
I've deleted this now (or rather I'm throwing an exception, but might want to delete that too), and instead try get the planner to use an underlying KEYWORD sub-field, if it exists, using fa.exactAttribute().